aleph-im/aleph-vm
# Fix 199 ruff errors

Explain what problem this PR is resolving

The python linter [ruff](https://github.com/astral-sh/ruff) is not enable on this project because it raises too many errors to fix at once. At the time of writing, `ruff check src` raises 375 errors.

## Self proofreading checklist

Not relevant for these changes.


## Changes

The first commit contains the result of `ruff check src --fix`, which contains safe automatic fixes to the codebase.

I then ran `ruff check src --fix --unsafe-fixes` and added the following commits with the changes that I reviewed manually and deem ready to commit:
- Fix: Generator loop -> 'yield from'
- Fix: Type annotations could be improved
- Fix: Python < 3.10 is no supported anymore
- Fix: Ruff errored on exception raising

## How to test

Run `ruff check src` or `hatch run testing:ruff`.

`ruff check src` on the code found 176 errors after the changes.
